### PR TITLE
Fix resource exhaustion security vulnerability

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Fix resource exhaustion security vulnerability

### CODE REVIEW

File: components/event-service/internal/externalapi/genericeventhandler.go [hunk 1]
```diff
@@ -41,9 +45,15 @@ type traceHeaderHandler struct {
...
-			r = r.WithContext(cloudevents.ContextWithHeader(r.Context(), key, r.Header.Get(key)))
+	for _, traceHeader := range traceHeaders {
+		xb3HeaderKey := fmt.Sprintf("%s%s", xB3Prefix, traceHeader)
+		if val := r.Header.Get(xb3HeaderKey); len(val) > 0 {
+			r = r.WithContext(cloudevents.ContextWithHeader(r.Context(), xb3HeaderKey, r.Header.Get(xb3HeaderKey)))
+		}
+
+		b3HeaderKey := fmt.Sprintf("%s%s", b3Prefix, traceHeader)
+		if val := r.Header.Get(b3HeaderKey); len(val) > 0 {
+			r = r.WithContext(cloudevents.ContextWithHeader(r.Context(), b3HeaderKey, r.Header.Get(b3HeaderKey)))
```
Comment (k15r) [inline components/event-service/internal/externalapi/genericeventhandler.go:56]: How about something like this:
```
	for _, traceHeader := range traceHeaders {
	    for _, HeaderKeyPrefix := range traceHeaderPrefixes {
	        fmt.Sprintf("%s%s", HeaderKeyPrefix, traceHeader)
	        if val := r.Header.Get(xb3HeaderKey); len(val) > 0 {
                    r = r.WithContext(cloudevents.ContextWithHeader(r.Context(), xb3HeaderKey, r.Header.Get(xb3HeaderKey)))
                 }
	    }
        }
```

Comment (sayanh) [reply components/event-service/internal/externalapi/genericeventhandler.go]: Fixed

**Review by k15r — APPROVED** [review]

**Pull request merged**